src/Storages/MergeTree/KeeperAccess.h

```diff
@@ -0,0 +1,236 @@
+#pragma once
+#include <random>
+#include <Common/ZooKeeper/IKeeper.h>
+#include <Common/ZooKeeper/KeeperException.h>
+#include <Common/ZooKeeper/Types.h>
+#include <Common/ZooKeeper/ZooKeeper.h>
+#include <Common/randomSeed.h>
+
+namespace DB
+{
+
+class FaultInjection
```

Comment:
Yes, I had 2 implementations initially. But finally decided to keep only one for now because another implementation (`FailEachCallOnce`) would require `KeeperAccess` and retries controllers to work together. Which I decided to not do for now.